src/Packetery/Module/Checkout.php

```diff
@@ -1217,4 +1216,36 @@ private function getPickupPointValidateRequest(
 		);
 	}
 
+	/**
+	 * Validates if payment method can be used with current cart.
+	 *
+	 * @param Carrier\Options $carrierOptions Carrier options.
+	 *
+	 * @return bool
+	 */
+	public function validateMaximumCod( Carrier\Options $carrierOptions ): bool {
+		if (
+			null === $carrierOptions->getMaximumCodValue() ||
+			0.0 === $carrierOptions->getMaximumCodValue()
+		) {
+			return true;
+		}
+
+		$cartPrice  = $this->getCartContentsTotalIncludingTax();
+		$cartWeight = $this->getCartWeightKg();
+		$cost       = $this->getRateCost( $carrierOptions, $cartPrice, $cartWeight );
+		// TODO: fees can't be determined at this time. Provide new filter for the user?
```

Comment:
Hádám, že metoda se volá po potvrzení údajů v pokladně zákazníkem. Pokladna v takovém případě ví výši poplatků. Šlo by poplatky vzít z odeslaných formulářových polí a nebo ze session?

Osobně bych zkusil překalkulovat košík. Tuším metodu "calculate_shipping".

Pokud se logika zacyklí, tak bych zkoumal proč a jestli to někde nejde lépe opodmínkovat.

V nejhoším případě by to mělo jít předat Javascriptem v pokladně. Ale to by bylo ošklivé.